apollo-federation/src/sources/connect/validation/selection.rs

```diff
@@ -96,13 +114,59 @@ pub(super) fn validate_body_selection(
     }
 
     // TODO: validate JSONSelection
+
+    if let Some(value) = validate_selection_variables(
+        ConnectorsContext::new(connect_coordinate.into(), Phase::Request, Target::Body),
+        schema,
+        selection_node,
+        coordinate,
+        &selection,
+    ) {
+        return value;
+    }
+
     Ok(())
 }
 
+/// Validate variable references in a JSON Selection
+fn validate_selection_variables(
+    expression_context: ConnectorsContext,
+    schema: &SchemaInfo,
+    selection_node: &Node<Value>,
+    coordinate: String,
+    selection: &JSONSelection,
+) -> Option<Result<(), Message>> {
+    let namespaces: HashSet<Namespace> = expression_context.available_namespaces().collect();
+    for reference in selection
+        .external_var_paths()
+        .into_iter()
+        .flat_map(|var_path| var_path.variable_reference())
+    {
+        if !namespaces.contains(&reference.namespace.namespace) {
```

Comment:
this doesn't yet use `VariableResolver` to validate the variable's path, right?

Reply:
Good call, I'll add that.